Sources/AardvarkReveal/RevealAttachmentGenerator.swift

```diff
@@ -0,0 +1,518 @@
+///   │                                                           │
+///   │  .─────────────────────────────────────────────────────.  │
+///   │ ( revealAttachmentGeneratorDidCaptureMainScreenSnapshot ) │
+///   │  `─────────────────────────────────────────────────────'  │
+///   └───────────────────────────────────────────────────────────┘
+///                                 │
+///                                 ▼
+///   ┌───────────────────────────────────────────────────────────┐
+///   │                                                           │
+///   │                   Bundling Reveal File                    │
+///   │                                                           │
+///   └───────────────────────────────────────────────────────────┘
+///                                 │
+///    .────────────────────────────┴────────────────────────────.
+///   (   revealAttachmentGeneratorDidFinishBundlingRevealFile    )
+///    `────────────────────────────┬────────────────────────────'
+///                                 ▼
+///                   .───────────────────────────.
+///                  (         completion          )
+///                   `───────────────────────────'
+/// ```
+///
+/// It is important to avoid changing anything while the app state is being captured (during the `Capturing App State`
+/// phase). Once the app state is captured, the generator will begin capturing snapshots of each drawable object (views,
+/// layers, etc.) on the screen. Once the generator has captured a snapshot of the main screen (marked by a call to the
+/// delegate's `revealAttachmentGeneratorDidCaptureMainScreenSnapshot` method), it is okay to display a waiting
+/// indicator **as long as all of the existing views/layers remain in the hierarchy** (for example, you can display a
+/// new window on top of the main window or present a modal view controller, depending on the use case). Once the
+/// generator has completed bundling the Reveal file, user interaction can be restored.
+///
+/// # Security Settings
+///
+/// The generator communicates with the Reveal server over HTTP. By default, the App Transport Security policy requires
+/// the use of a secure connection, which is not available. To enable the generator to connect to the server, add the
+/// following to your app's `Info.plist`:
+///
+/// ```xml
+/// <key>NSAppTransportSecurity</key>
+/// <dict>
+///     <key>NSExceptionDomains</key>
+///     <dict>
+///         <key>localhost</key>
+///         <dict>
+///             <key>NSExceptionAllowsInsecureHTTPLoads</key>
+///             <true/>
+///             <key>NSIncludesSubdomains</key>
+///             <true/>
+///         </dict>
+///     </dict>
+/// </dict>
+/// ```
+@objc(ARKRevealAttachmentGenerator)
+public final class RevealAttachmentGenerator: NSObject {
+
+    // MARK: - Life Cycle
+
+    public override init() {
+        super.init()
+
+        serviceBrowser.startSearching()
```

Comment:
😬 why are we kicking off work from `init`?

Reply:
Fair question. I was trying to make it hard to get wrong (i.e. if you forget to start it early). Maybe that's too magical and I should expose the start/stop methods.